tests/sqlparser_clickhouse.rs

```diff
@@ -1646,6 +1646,16 @@ fn parse_table_sample() {
     clickhouse().verified_stmt("SELECT * FROM tbl SAMPLE 1 / 10 OFFSET 1 / 2");
 }
 
+#[test]
+fn parse_numbers_with_underscore() {
+    let select = clickhouse().verified_only_select("SELECT 10_000");
```

Comment:
@graup regarding the test failure, its due to the bigdecimal representation when the feature flag it enabled, could you try something like this for the assertion?
```rust
let canonical = if cfg!(feature = "bigdecimal") {
    "SELECT 10000"
} else {
    "SELECT 10_000"
};
let select = clickhouse().one_statement_parses_to(
    "SELECT 10_000",
    canonical,
);
// ...
```